packages/@aws-cdk/aws-servicecatalog/lib/private/product-stack-synthesizer.ts

```diff
@@ -16,17 +20,28 @@ export class ProductStackSynthesizer extends cdk.StackSynthesizer {
   }
 
   public addFileAsset(_asset: cdk.FileAssetSource): cdk.FileAssetLocation {
-    throw new Error('Service Catalog Product Stacks cannot use Assets');
+    if (!this.stack) {
+      throw new Error('You must call bindStack() first');
+    }
+
+    if (!this.assetBucket) {
+      const parentStack = (this.stack as ProductStack)._getParentStack();
+      this.assetBucket = new ProductStackAssetBucket(parentStack, `ProductStackAssetBucket${hashValues(this.stack.stackName)}`);
```

Comment:
Hashing this doesn't seem necessary. Why not have the product name here?

Reply:
You've picked the scope of asset buckets to be individual ProductStacks. What's the reason for picking that over, say, an asset bucket per Portfolio? Given that the entire portfolio is shared with the same groups of accounts anyway, a bucket per portfolio seems more desirable?

Reply:
> You've picked the scope of asset buckets to be individual ProductStacks. What's the reason for picking that over, say, an asset bucket per Portfolio? Given that the entire portfolio is shared with the same groups of accounts anyway, a bucket per portfolio seems more desirable?

Check out this comment: https://github.com/aws/aws-cdk/pull/21508#discussion_r942536461

Reply:
> Hashing this doesn't seem necessary. Why not have the product name here?

Let's take a look at the following scenario:
```
    const product = new sc.CloudFormationProduct(this, 'HelloServerless', {
      productName: "HelloServerless",
      owner: "IT",
      productVersions: [
        {
          cloudFormationTemplate: sc.CloudFormationTemplate.fromProductStack(new ProductStack1(this, 'ProductStack1'),
          productVersionName: "v1",
          description: "A simple REST API backed by a hello Lambda function"
        },
        {
          cloudFormationTemplate: sc.CloudFormationTemplate.fromProductStack(new ProductStack2(this, 'ProductStack2')
          ),
          productVersionName: "v2",
          description: "A simple REST API backed by a goodbye Lambda function"
        },
      ],
    });
```

If we hashed by product name, the product stack asset buckets for both `ProductStack1` and `ProductStack2` would share the same id and thus error out. To avoid this, we can hash by parent stack + product stack which is what `this.stack.StackName` will give us.